Adding add token signing certificate content to v1 (#14137)
microsoftgraph/microsoft-graph-docs: api-reference/v1.0/resources/selfsignedcertificate.md

---
title: "selfSignedCertificate resource type"
description: "Contains information about the public part of a signing certificate."
ms.localizationpriority: medium
doc_type: resourcePageType
ms.prod: "applications"

> author: "davidmu"

Reviewer: ```suggestion
author: "luleonpla"
```

Author: I changed it to have myself as the owner since Luis is no longer in this area

---

# selfSignedCertificate resource type

Namespace: microsoft.graph

Contains the public part of a signing certificate. It's the return type of the action [addSelfSignedSigningCertificate](../api/serviceprincipal-addtokensigningcertificate.md). Service providers use the public part of the signing certificate to validate the issuer of the token.

## Properties
Property|Type|Description
----|--|---
|customKeyIdentifier|Binary| Custom key identifier. |
| displayName | String | The friendly name for the key. |

> |endDateTime|DateTimeOffset|The date and time at which the credential expires. The Timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 would look like this: "2014-01-01T00:00:00Z". |
> |keyId|Guid|The unique identifier (GUID) for the key.|
> |startDateTime|DateTimeOffset|The date and time at which the credential becomes valid. The Timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 would look like this: "2014-01-01T00:00:00Z". |
> |type|String|The type of key credential. "AsymmetricX509Cert".|
> |usage|String|A string that describes the purpose for which the key can be used. For example, "Verify".|
> |key|Binary| The value for the key credential. Should be a base-64 encoded value. |

Reviewer: ```suggestion
|endDateTime|DateTimeOffset|The date and time at which the credential expires. The timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 is `2014-01-01T00:00:00Z`.|
|keyId|Guid|The unique identifier (GUID) for the key.|
|startDateTime|DateTimeOffset|The date and time at which the credential becomes valid. The timestamp type represents date and time information using ISO 8601 format and is always in UTC time. For example, midnight UTC on Jan 1, 2014 is `2014-01-01T00:00:00Z`. |
|type|String|The type of key credential. "AsymmetricX509Cert".|
|usage|String|A string that describes the purpose for which the key can be used. The possible value is `Verify`.|
|key|Binary| The value for the key credential. Should be a Base-64 encoded value. |
```

|thumbprint| String | The thumbprint value for the key.|

## JSON representation

Here is a JSON representation of the resource

<!-- {
  "blockType": "resource",
  "optionalProperties": [

  ],
  "@odata.type": "microsoft.graph.selfSignedCertificate"
}-->

```json

> {
>     "customKeyIdentifier": "string (binary)",
>     "displayName": "string",
>     "endDateTime": "string (timestamp)",
>     "key": "string (binary)",
>     "keyId": "guid",
>     "startDateTime": "String (timestamp)",
>     "type": "string",
>     "thumbprint":"string",
>     "usage": "string"
> }

Reviewer: ```suggestion
``` json
{
  "@odata.type": "#microsoft.graph.selfSignedCertificate",
  "customKeyIdentifier": "String (Binary)",
  "displayName": "String",
  "endDateTime": "String (timestamp)",
  "key": "String (Binary)",
  "keyId": "Guid",
  "startDateTime": "String (timestamp)",
  "thumbprint": "String",
  "type": "String",
  "usage": "String"
}
```
